godot-core/src/builtin/basis.rs

```diff
@@ -0,0 +1,980 @@
+    ///
+    /// _Godot equivalent: `Basis.tdotz()`_
+    #[must_use]
+    pub fn tdotz(&self, with: Vector3) -> f32 {
+        self.col_z().dot(with)
+    }
+
+    /// Returns `true` if this basis is finite. Meaning each element of the
+    /// matrix is not `NaN`, positive infinity, or negative infinity.
+    ///
+    /// _Godot equivalent: `Basis.is_finite()`_
+    #[must_use]
+    pub fn is_finite(&self) -> bool {
+        self[0].is_finite() && self[1].is_finite() && self[2].is_finite()
+    }
+
+    /// Returns `true` if this basis and `other` are approximately equal,
+    /// by calling `is_equal_approx` on each row.
+    ///
+    /// _Godot equivalent: `Basis.is_equal_approx(Basis b)`_
+    pub fn is_equal_approx(&self, other: &Self) -> bool {
+        self[0].is_equal_approx(other[0])
+            && self[1].is_equal_approx(other[1])
+            && self[2].is_equal_approx(other[2])
+    }
+
+    /// Returns the first column of the matrix,
+    ///
+    /// _Godot equivalent: `Basis.x`_
+    #[must_use]
+    pub fn col_x(&self) -> Vector3 {
+        Vector3::new(self[0].x, self[1].x, self[2].x)
+    }
+
+    /// Set the values of the first column of the matrix.
+    pub fn set_col_x(&mut self, col: Vector3) {
+        self[0].x = col.x;
+        self[1].x = col.y;
+        self[2].x = col.z;
+    }
+
+    /// Returns the second column of the matrix,
+    ///
+    /// _Godot equivalent: `Basis.y`_
+    #[must_use]
+    pub fn col_y(&self) -> Vector3 {
+        Vector3::new(self[0].y, self[1].y, self[2].y)
+    }
+
+    /// Set the values of the second column of the matrix.
+    pub fn set_col_y(&mut self, col: Vector3) {
+        self[0].y = col.x;
+        self[1].y = col.y;
+        self[2].y = col.z;
+    }
+
+    /// Returns the third column of the matrix,
+    ///
+    /// _Godot equivalent: `Basis.z`_
+    #[must_use]
+    pub fn col_z(&self) -> Vector3 {
+        Vector3::new(self[0].z, self[1].z, self[2].z)
+    }
+
+    /// Set the values of the third column of the matrix.
+    pub fn set_col_z(&mut self, col: Vector3) {
+        self[0].z = col.x;
+        self[1].z = col.y;
+        self[2].z = col.z;
+    }
+}
+
+impl Display for Basis {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Godot output:
+        // [X: (1, 0, 0), Y: (0, 1, 0), Z: (0, 0, 1)]
+        // Where X,Y,Z are the columns
+        let [Vector3 {
+            x: x1,
+            y: x2,
+            z: x3,
+        }, Vector3 {
+            x: y1,
+            y: y2,
+            z: y3,
+        }, Vector3 {
+            x: z1,
+            y: z2,
+            z: z3,
+        }] = self.to_cols();
+
+        f.write_fmt(format_args!(
+            "[X: ({x1}, {x2}, {x3}), Y: ({y1}, {y2}, {y3}), Z: ({z1}, {z2}, {z3})]"
+        ))
+    }
+}
+
+impl From<Quaternion> for Basis {
+    fn from(quat: Quaternion) -> Self {
+        Basis::from_quat(quat)
+    }
+}
```

Comment:
I don't see a reason to allow `quat.into()` when we have a more explicit `Basis::from_quat()`. Basis and quaternion are conceptually different, even if they share a lot of properties.